master/buildbot/data/buildrequests.py

```diff
@@ -143,8 +143,10 @@ def get(self, resultSpec, kwargs):
             claimed=claimed,
             bsid=bsid,
             resultSpec=resultSpec)
-        defer.returnValue(
-            [(yield self.db2data(br)) for br in buildrequests])
+        results = []
+        for br in buildrequests:
+            results.append((yield self.db2data(br)))
+        defer.returnValue(results)
```

Comment:
I got it to work with this:

        steps_dict = [self.db2data(dbdict) for dbdict in steps] 
        list_results = yield defer.DeferredList(steps_dict)
        results = [res[1] for res in list_results]
        defer.returnValue(results)

Not quite what I had in mind. It does allow the `self.db2data` calls to run concurrently but isn't that pretty.